bindings/python/test/test_arrow.py

```diff
@@ -325,6 +327,65 @@ def test_string_bool(self):
             ),
         )
 
+    def test_auto_schema(self):
+        # Create table with random data of various types.
+        data = Table.from_pydict(
+            {
+                "string": [None] + [str(i) for i in range(2)],
+                "bool": [True for _ in range(3)],
+                "dt": [datetime(1970 + i, 1, 1) for i in range(3)],
+            },
+            # Ordering of output auto-schema will be the first fields detected.
+            # Test will fail if not ordered correctly.
```

Comment:
Is this comment only needed because the order of the fields above differs from the schema below?

Reply:
The columns of the auto-detected schema is in the order of when a field first appears in the stream, so `bool` and `dt` will get detected first, and then `string` third. I had the schema ordered differently before, and it would on `self.assertEqual(data, out)` because the orderings of the schema and the detected schema did not match. I'll also the data dictionary change it to match.

Reply:
I still have the same question. If we change the from_pydict and ArrowSchema key order to match, then I believe this comment is no longer needed.